redis_sentinel/check.py

```diff
@@ -0,0 +1,235 @@
+
+
+class RedisSentinelCheck(AgentCheck):
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+        self._masters = defaultdict(lambda: "")
+
+    def check(self, instance):
+        self.log.info(instance)
+        redis_conn = redis.StrictRedis(
+            host=instance['sentinel_host'], port=int(instance['sentinel_port']), db=0
+        )
+        for master_name in instance['masters']:
+            base_tags = ['redis_name:%s' % master_name] + instance.get('tags', [])
+            try:
+                self._process_instance_master(redis_conn, master_name, base_tags)
+            except Exception as e:
+                self.warning("Error collecting metrics for master %s: %s" % (master_name, e))
+
+    def _process_instance_master(self, redis_conn, master_name, base_tags):
+        master_tags = self._process_master_stats(redis_conn, master_name, base_tags)
+        self._process_slaves_stats(redis_conn, master_name, base_tags, master_tags)
+        self._process_sentinels_stats(redis_conn, master_name, base_tags, master_tags)
+
+    def _process_sentinels_stats(self, redis_conn, master_name, base_tags, master_tags):
+        """
+            [{
+                'down-after-milliseconds': 5000,
+                'flags': 's_down,sentinel',
+                'ip': '10.1.2.3',
+                'is_disconnected': False,
+                'is_master': False,
+                'is_master_down': False,
+                'is_odown': False,
+                'is_sdown': True,
+                'is_sentinel': True,
+                'is_slave': False,
+                'last-hello-message': 12345678,
+                'last-ok-ping-reply': 12345678,
+                'last-ping-reply': 12345679
+                'last-ping-sent': 12345678,
+                'name': '10.1.2.3:26379',
+                'pending-commands': 78,
+                'port': 26379,
+                'runid': '123456789abcdef',
+                's-down-time': 12345678,
+                'voted-leader': '?',
+                'voted-leader-epoch': 0,
+            }]
+        """
+        sentinels_stats = redis_conn.sentinel_sentinels(master_name)
+        # sentinel_stats returns stats for other sentinels only
+        # so increment once for current sentinel
+        self.increment('redis.sentinel.ok_sentinels', tags=master_tags)
+        for stats in sentinels_stats:
+            sentinel_tags = ['sentinel_ip:%s' % stats['ip']] + base_tags
+            if stats['is_odown'] or stats['is_sdown']:  # sentinel keeps track of old sentinels
+                continue
+            self.increment('redis.sentinel.ok_sentinels', tags=master_tags)
+
+            pending = stats.get('pending-commands', stats.get('link-pending-commands'))
+            if pending is not None:
+                self.gauge(
+                    'redis.sentinel.pending_commands',
```

Comment:
Do you think it would make more sense if this was called `link_pending_commands` instead? I think it would be beneficial to be consistent with latest redis-provided name `link-pending-commands`